xbmc/addons/AddonManager.cpp

```diff
@@ -1190,7 +1192,19 @@ bool CAddonMgr::AddonsFromRepoXML(const CRepository::DirInfo& repo, const std::s
             info->identifier, info->identifier, builder.GetVersion().asString().c_str())));
         auto addon = builder.Build();
         if (addon)
-          addons.push_back(std::move(addon));
+        {
+          if (repoId.empty() || repoId == OVERRIDE_REPOSITORY)
+            addons.push_back(std::move(addon));
+          else
+          {
+            ADDON::VECADDONS addonsFind;
+            m_database.FindByAddonId(addon.get()->ID(), addonsFind);
+            if (std::find_if(addonsFind.begin(), addonsFind.end(), [](AddonPtr const& p) { return p.get()->Origin() == OVERRIDE_REPOSITORY; }) == addonsFind.end())
+              addons.push_back(std::move(addon)); 
+            else
+              CLog::Log(LOGNOTICE, "CAddonMgr: %s served by %s, skiping version from %s", addon.get()->ID().c_str(), OVERRIDE_REPOSITORY.c_str(), repoId.c_str());
+          }
+        }
```

Comment:
This is all wrong anyway. Updates does not happen in any order so there is no guarantee anything will exist in the db. And it makes the function do different things depending on mutable global state; ugh

Reply:
I would expect database already has information from override repo, if you install a 3rd party repo.
Also I know that it may compares with the data from previous sync from the override repo, but I don't expect it would be that tragic.

Reply:
@tamland just a general question, would you suggest me to not filter the addons inserted into the db, but filter the queries like I've done in variant 1?